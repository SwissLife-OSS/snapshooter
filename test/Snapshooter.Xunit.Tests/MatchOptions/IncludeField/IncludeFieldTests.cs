using Snapshooter.Tests.Data;
using Xunit;

namespace Snapshooter.Xunit.Tests.MatchOptions.IncludeField;

public class IncludeFieldTests
{
    [Fact]
    public void IncludeField_StringScalarField_IncludedOnlyField()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Firstname"));
    }

    [Fact]
    public void IncludeField_StringScalarFields_IncludedOnlyFields()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Firstname")
            .IncludeField("Age")
            .IncludeField("CreationDate"));
    }

    [Fact]
    public void IncludeField_StringDuplicatedScalarFields_IncludedOnlyFields()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Age")
            .IncludeField("Lastname")
            .IncludeField("Age")
            .IncludeField("**.Lastname"));
    }
    
    [Fact]
    public void IncludeField_ComplexObjectField_IncludedOnlyField()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Address"));
    }

    [Fact]
    public void IncludeField_IncludeTwiceInPath_RightFieldIncluded()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Address")
            .IncludeField("Address.Country"));
    }

    [Fact]
    public void IncludeField_IncludeTwoDifferentFields_RightFieldIncluded()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Children")
            .IncludeField("Relatives"));
    }

    [Fact]
    public void IncludeField_IncludeFieldsByName_IncludedOnlyFields()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("**.Address")
            .IncludeField("**.Name")
            .IncludeField("**.Country")
            .IncludeField("**.Lastname"));
    }

    [Fact]
    public void IncludeField_IncludeArrayFields_IncludedOnlyFields()
    {
        // arrange
        TestPerson testPerson = TestDataBuilder
            .TestPersonMarkWalton()
            .Build();

        // act & assert
        Snapshot.Match(testPerson, options => options
            .IncludeField("Children[2]")
            .IncludeField("Relatives[0]"));
    }
}
