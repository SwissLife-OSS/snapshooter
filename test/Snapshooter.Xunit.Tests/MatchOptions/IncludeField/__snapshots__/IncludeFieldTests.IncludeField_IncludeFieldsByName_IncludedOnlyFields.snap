{
  "Lastname": "Walton",
  "Address": {
    "Street": "Rohrstrasse",
    "StreetNumber": 12,
    "Plz": 8304,
    "City": "Wallislellen",
    "Country": {
      "Name": "Switzerland",
      "CountryCode": "CH"
    }
  },
  "Children": [
    {
      "Name": "James"
    },
    {
      "Name": null
    },
    {
      "Name": "Hanna"
    }
  ],
  "Relatives": [
    {
      "Lastname": "Schneider",
      "Address": {
        "Street": "Bahnhofstrasse",
        "StreetNumber": 450,
        "Plz": 8000,
        "City": "Zurich",
        "Country": {
          "Name": "Switzerland",
          "CountryCode": "CH"
        }
      }
    }
  ]
}
